src/open_weather_map/client.rs

```diff
@@ -43,6 +44,36 @@ impl WeatherApiClient {
         Self::decode_response(response)
     }
 
+    pub async fn fetch_weekly(&self, lat: f64, lon: f64) -> Result<WeatherForecast, ClientError> {
+        let request_url = format!(
+            "https://api.openweathermap.org/data/2.5/forecast?lat={}&lon={}&appid={}&units={}&lang={}&cnt={}",
+            lat,
+            lon,
+            OPEN_WEATHER_MAP_API_TOKEN.as_str(),
+            UNITS,
+            LANG,
+            9,
+        );
+
+        let response = self.client.get(&request_url).send()?;
+
+        Self::decode_weekly_response(response)
+    }
+
+    pub fn decode_weekly_response(
```

Comment:
can be private

Reply:
[Make private decodes and deleting no longer more useful test](https://github.com/pxp9/weather_bot_rust/pull/29/commits/15b1bee39f8105061f5757fec30dbfdcf6bb2433)